.github/workflows/build-ubuntu.yml

```diff
@@ -0,0 +1,272 @@
+            libceres-dev \
+            libflann-dev \
+            libfreeimage-dev \
+            libmetis-dev \
+            libgoogle-glog-dev \
+            libgtest-dev \
+            libsqlite3-dev \
+            libglew-dev \
+            qtbase5-dev \
+            libqt5opengl5-dev \
+            libcgal-dev \
+            libcgal-qt5-dev \
+            libgl1-mesa-dri \
+            libunwind-dev \
+            xvfb
+
+          if [ "${{ matrix.config.cudaEnabled }}" == "true" ]; then
+            if [ "${{ matrix.config.os }}" == "ubuntu-20.04" ]; then
+              sudo apt-get install -y \
+                nvidia-cuda-toolkit \
+                nvidia-cuda-toolkit-gcc
+              echo "CC=/usr/bin/cuda-gcc" >> $GITHUB_ENV
+              echo "CXX=/usr/bin/cuda-g++" >> $GITHUB_ENV
+            elif [ "${{ matrix.config.os }}" == "ubuntu-22.04" ]; then
+              sudo apt-get install -y \
+                nvidia-cuda-toolkit \
+                nvidia-cuda-toolkit-gcc \
+                gcc-10 g++-10
+              echo "CC=/usr/bin/gcc-10" >> $GITHUB_ENV
+              echo "CXX=/usr/bin/g++-10" >> $GITHUB_ENV
+              echo "CUDAHOSTCXX=/usr/bin/g++-10" >> $GITHUB_ENV
+            fi
+          fi
+
+          if [ "${{ matrix.config.asanEnabled }}" == "true" ]; then
+            sudo apt-get install -y clang-15 libomp-15-dev
+            echo "CC=/usr/bin/clang-15" >> $GITHUB_ENV
+            echo "CXX=/usr/bin/clang++-15" >> $GITHUB_ENV
+          fi
+
+          if [ "${{ matrix.config.cmakeBuildType }}" == "ClangTidy" ]; then
+            sudo apt-get install -y clang-15 clang-tidy-15 libomp-15-dev
+            echo "CC=/usr/bin/clang-15" >> $GITHUB_ENV
+            echo "CXX=/usr/bin/clang++-15" >> $GITHUB_ENV
+          fi
+
+      - name: Configure and build
+        run: |
+          set -x
+          cmake --version
+          mkdir build
+          cd build
+          cmake .. \
+            -GNinja \
+            -DCMAKE_BUILD_TYPE=${{ matrix.config.cmakeBuildType }} \
+            -DCMAKE_INSTALL_PREFIX=./install \
+            -DCMAKE_CUDA_ARCHITECTURES=50 \
+            -DTESTS_ENABLED=ON \
+            -DCUDA_ENABLED=${{ matrix.config.cudaEnabled }} \
+            -DGUI_ENABLED=${{ matrix.config.guiEnabled }} \
+            -DASAN_ENABLED=${{ matrix.config.asanEnabled }}
+          ninja -k 10000
+
+      - name: Install and build sample
+        run: |
+          if [ "${{ matrix.config.asanEnabled }}" == "true" ] ||
+             [ "${{ matrix.config.cmakeBuildType }}" == "ClangTidy" ]; then
+            exit 0
+          fi
+
+          set -x
+          cd build
+          ninja install
+          cd ../doc/sample-project
+          mkdir build
+          cd build
+          colmap_DIR=${{ github.workspace }}/build/install/share/colmap cmake .. \
+            -GNinja \
+            -DCMAKE_CUDA_ARCHITECTURES=50
+          ninja
+          ./hello_world --message "world"
+
+      - name: Run tests
```

Comment:
```suggestion
      - name: Run unit tests
```

Reply:
I'll keep it as is, since these are not just unit tests but also integration tests on the controller level with synthetic data, etc.